datetime/datetime_test.go

```diff
@@ -0,0 +1,536 @@
+	if err != nil || resp.Code != 0 {
+		t.Fatalf("Failed to replace: %s", err.Error())
+		return
+	}
+	if len(resp.Data) != 1 {
+		t.Fatalf("Response Body len != 1")
+	}
+	if tpl, ok := resp.Data[0].([]interface{}); !ok {
+		t.Errorf("Unexpected body of Replace")
+	} else {
+		if len(tpl) != 3 {
+			t.Fatalf("Unexpected body of Replace (tuple len)")
+		}
+		if id, ok := tpl[0].(uint64); !ok || id != 13 {
+			t.Fatalf("Unexpected body of Replace (13)")
+		}
+		if o, ok := tpl[1].(string); !ok || o != "orig" {
+			t.Fatalf("Unexpected body of Replace (1)")
+		}
+	}
+}
+
+func TestCustomEncodeDecodeTuple2(t *testing.T) {
+	if isDatetimeSupported == false {
+		t.Skip("Skipping test for Tarantool without datetime support in msgpack")
+	}
+
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	// Setup: insert a value.
+	tm := time.Unix(0, 0)
+	_, err := conn.Insert(space, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime insert failed: %s", err.Error())
+	}
+
+	var tuples []Tuple
+	err = conn.SelectTyped("testDatetime_2", index, 0, 1, IterEq, []interface{}{1, tm}, &tuples)
+	if err != nil {
+		t.Fatalf("Failed to SelectTyped(): %s", err.Error())
+	}
+
+	// Teardown: delete a value.
+	_, err = conn.Delete(space, index, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime delete failed: %s", err.Error())
+	}
+}
+
+func TestCustomEncodeDecodeTuple3(t *testing.T) {
+	if isDatetimeSupported == false {
+		t.Skip("Skipping test for Tarantool without datetime support in msgpack")
+	}
+
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	// Setup: insert a value.
+	tm := time.Unix(0, 0)
+	_, err := conn.Insert(space, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime insert failed: %s", err.Error())
+	}
+
+	var tuples2 []Tuple2
+	err = conn.SelectTyped("testDatetime_2", index, 0, 1, IterEq, []interface{}{1, tm}, &tuples2)
+	if err != nil {
+		log.Fatalf("Failed to SelectTyped: %s", err.Error())
+		return
+	}
+
+	// Teardown: delete a value.
+	_, err = conn.Delete(space, index, []interface{}{tm})
+	if err != nil {
+		t.Fatalf("Datetime delete failed: %s", err.Error())
+	}
+}
+
+func TestCustomEncodeDecodeTuple4(t *testing.T) {
+	if isDatetimeSupported == false {
+		t.Skip("Skipping test for Tarantool without datetime support in msgpack")
+	}
+
+	conn := connectWithValidation(t)
+	defer conn.Close()
+
+	// Setup: insert values.
+	_, err := conn.Insert(space, []interface{}{time.Unix(10, 0)})
+	if err != nil {
+		t.Fatalf("Datetime insert failed: %s", err.Error())
+	}
+	_, err = conn.Insert(space, []interface{}{time.Unix(11, 0)})
+	if err != nil {
+		t.Fatalf("Datetime insert failed: %s", err.Error())
+	}
+	_, err = conn.Insert(space, []interface{}{time.Unix(12, 0)})
+	if err != nil {
+		t.Fatalf("Datetime insert failed: %s", err.Error())
+	}
+
+	// Call function 'call_me_maybe' returning a table of custom tuples.
+	var tuples []Tuple2
+	err = conn.CallTyped("call_me_maybe", []interface{}{1}, &tuples)
```

Comment:
I don't get it. There are no tuples and datetime objects, only Lua tables, strings and numbers

Reply:
I don't get your question.

Reply:
Why do we test it here? I don't see any datetime-related info in `call_me_maybe` procedure.

Reply:
Yura, asked me to add these tests. See his comment above (in Russian).

Reply:
I agree, these tests do not add value to testing datetime support itself.